Services/source/updater/updater.c

```diff
@@ -30,41 +49,36 @@ SAT_returnState updater_app(csp_packet_t *packet) {
     uint8_t ser_subtype = (uint8_t)packet->data[SUBSERVICE_BYTE];
     int32_t fp;
     int8_t status;
-    uint8_t *buf = "Hello worldddddddd";
-    image_info app_info;
-
-    switch (ser_subtype) {
-      case FLASH_UPDATE:
-          fp = red_open("/application_image.bin", RED_O_RDONLY);
-          if (fp == -1) {
-              status = -1; break;
-          }
-          uint8_t test;
-          red_read(fp, &test, 1);
-          REDSTAT file_info;
-          int res = red_fstat(fp, &file_info);
-          if (res == -1) {
-              int err = red_errno;
-              status = -1; break;
-          }
-          if (!init_eeprom()) {
-              status = -1; break;
-          }
-
-          app_info = priv_eeprom_get_app_info();
-          if (!BLInternalFlashStartAddrCheck(app_info.addr, (uint32_t)file_info.st_size)){
-              status = -1; break;
-          }
+    uint8_t *buf;
+
+    if (init_eeprom()) {
+        switch (ser_subtype) {
+          case FLASH_UPDATE:
+    #ifdef GOLDEN_IMAGE
+              fp = red_open("VOL0:/application_image.bin", RED_O_RDONLY);
+              if (fp == -1) {
+                  status = -1; break;
+              }
+              REDSTAT file_info;
+              int res = red_fstat(fp, &file_info);
+              if (res == -1) {
+                  int err = red_errno;
```

Comment:
Are we doing anything with this? If not lets ignore this perhaps. Also maybe log the failure.

Reply:
The errno thing can be removed, but yes the fstat is used elsewhere